tests/git-ssh-server/Earthfile

```diff
@@ -1,67 +1,354 @@
+VERSION 0.6
+
 ARG DOCKERHUB_USER_SECRET=+secrets/DOCKERHUB_USER
 ARG DOCKERHUB_TOKEN_SECRET=+secrets/DOCKERHUB_TOKEN
 ARG DOCKERHUB_MIRROR
 ARG DOCKERHUB_AUTH=true
-FROM ..+base \
-    --DOCKERHUB_AUTH=$DOCKERHUB_AUTH \
-    --DOCKERHUB_USER_SECRET=$DOCKERHUB_USER_SECRET \
-    --DOCKERHUB_TOKEN_SECRET=$DOCKERHUB_TOKEN_SECRET \
-    --DOCKERHUB_MIRROR=$DOCKERHUB_MIRROR
-
-# server creates a ssh-based git server; this repo can then be cloned by running
-# "git clone git@selfsigned.example.com:testuser/repo.git"
-server:
-    RUN apk add --update --no-cache openssh git
-    RUN ssh-keygen -A
-    RUN ssh-keygen -b 3072 -t rsa -f /root/self-hosted-sshkey -q -N "" -C "testkey"
-    RUN adduser --disabled-password --gecos "" git && \
-        mkdir ~git/.ssh && \
-        cat /root/self-hosted-sshkey.pub > ~git/.ssh/authorized_keys && \
-        chmod 600 ~git/.ssh/authorized_keys && \
-        chown git:git ~git/.ssh/authorized_keys && \
-        cat /etc/passwd | grep git && \
-        sed -i 's/\(git:.*\):\/bin\/ash/\1:\/usr\/bin\/git-shell/g' /etc/passwd && \
-        cat /etc/passwd | grep git && \
-        passwd git -u
-    RUN mkdir /home/git/testuser && \
-        git init --bar /home/git/testuser/repo.git && \
-        chown -R git:git /home/git/testuser
-    COPY sshd_config /etc/ssh/sshd_config
-    RUN echo "Welcome to our self-hosted git ssh server" > /etc/motd
-
-    RUN echo "set -e
-# Docker creates /etc/hosts as readonly; we must unmount it first
-# before we can update it; this change WILL not stick around between different RUNs
-# furthermore, we must execute this script with RUN --privileged
-cp /etc/hosts /tmp/hosts
-umount /etc/hosts
-cp /tmp/hosts /etc/hosts
-if ! grep git.example.com /etc/hosts 2>/dev/null; then
-    echo -e \"127.0.0.1\tgit.example.com\" >> /etc/hosts
+
+test-rsa-only:
+    FROM ./setup+server \
+        --DOCKERHUB_AUTH=$DOCKERHUB_AUTH \
+        --DOCKERHUB_USER_SECRET=$DOCKERHUB_USER_SECRET \
+        --DOCKERHUB_TOKEN_SECRET=$DOCKERHUB_TOKEN_SECRET \
+        --DOCKERHUB_MIRROR=$DOCKERHUB_MIRROR \
+        --SSHD_RSA=true \
+        --SSHD_ED25519=false \
+        --USER_RSA=true \
+        --USER_ED25519=false
+    RUN echo "#!/bin/sh
+set -ex
+
+# first ensure these two /etc/hosts entries are working
+ping -c 1 git.example.com
+ping -c 1 buildkitsandbox
+
+# load in preauthorized key
+eval \$(ssh-agent)
+ssh-add /root/self-hosted-rsa-key
+ssh-add -l
+
+earthly --config \$earthly_config --verbose -D +test
```

Comment:
What's the reason for having a local +test target? Is it testing ssh behavior still?

Reply:
Ah I see the GIT CLONE in git-clone-private-ssh.earth now